newton/_src/utils/mesh.py

```
@@ -261,6 +265,419 @@ def create_ellipsoid_mesh(
+    Returns:
+        List of Mesh objects.
+    """
+    import trimesh  # noqa: PLC0415
+
+    filename = os.fspath(filename)
+    scale = np.asarray(scale, dtype=np.float32)
+    base_dir = os.path.dirname(filename)
+
+    def _parse_dae_material_colors(path: str) -> tuple[list[str], dict[str, tuple[float, float, float]]]:
+        try:
+            tree = ET.parse(path)
```

Ruff (0.14.13)

Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)